src/providers/slot/slot.ts

```diff
@@ -99,30 +102,34 @@ export class SlotProvider {
     const { testPermissionPeriods } = this.appConfigProvider.getAppConfig().journal;
     const testCategory = get(testSlot, 'booking.application.testCategory');
     const startDate = new DateTime(testSlot.slotDetail.start);
+    const slotStartDate: Date = new Date(testSlot.slotDetail.start);
 
     if (!testCategory || startDate.daysDiff(this.dateTimeProvider.now()) < 0) {
       return false;
     }
 
     const periodsPermittingStart = testPermissionPeriods.filter((period) => {
-      const slotHasPeriodStartCriteria = this.hasPeriodStartCriteria(startDate, period.from);
-      const slotHasPeriodEndCriteria = this.hasPeriodEndCritiera(startDate, period.to);
+      const slotHasPeriodStartCriteria: boolean = this.hasPeriodStartCriteria(slotStartDate, period.from);
+      const slotHasPeriodEndCriteria: boolean = this.hasPeriodEndCriteria(slotStartDate, period.to);
       return period.testCategory === testCategory && slotHasPeriodStartCriteria && slotHasPeriodEndCriteria;
     });
     return periodsPermittingStart.length > 0;
   }
 
-  private hasPeriodStartCriteria(slotDate: DateTime, periodFrom: string) {
-    const periodStartDate = new DateTime(periodFrom);
-    return slotDate.daysDiff(periodStartDate) <= 0;
+  private dateDiffInDays = (startDate: Date, periodDate: Date): number => {
```

Comment:
Quick unit test to cover this new method?

Reply:
shall i make it public in order to test it?